### PR TITLE
Change the build process to use rollup.

### PR DESCRIPTION
**Pull request by realityking**
This PR includes the changes from #41.

Rollup is configured to do the absolute minimum amount of bundling. All dependencies are marked as external.

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,7 +1,10 @@
 {
   "presets": [
     "react",
-    ["es2015", { "loose": true }],
+    ["env", { "targets": {"node": "current"}, "loose": true }],
     "stage-2"
+  ],
+  "plugins": [
+    "add-module-exports"
```
Comment (lourd) [inline .babelrc]: It appears that just using the `react` and `env` presets will suffice. We can drop the `stage-2` preset, and I don't think we need the `add-module-exports` plugin.

**Review by lourd — CHANGES_REQUESTED** [review]
Hey @realityking, thanks for the changes! I have one note about the babel config, otherwise these look good to me.

**Comment by lourd** [comment]
Hey @realityking, friendly bump about this PR if you're still interested in getting it merged.

File: .babelrc (hunk 1 above)
Comment (realityking) [reply .babelrc]: There are two transforms used that are not part of `babel-preset-env`, I've added them individually as that is more self documenting (and IMHO safer) than the stage presets. The one for object rest/spread can be dropped with Babel7.

 add-module-exports is necessary as the tests don't run through rollup.

**Comment by realityking** [comment]
@lourd Sorry about that, completely dropped from my mind.

I've since earned some more experience with ES modules and think it's a bad idea to publish any that use default exports. The interoperability by default is not great when using Webpack and CommonJS.

I left all other changes in place as they still make sense, adding support for ESM is in the future a 2-line change.

**Review by lourd — APPROVED** [review]

**Comment by lourd** [comment]
No worries! This looks great. Thanks!

**Pull request merged**